src/main/java/org/jenkinsci/test/acceptance/po/SnippetGenerator.java

```diff
@@ -0,0 +1,40 @@
+package org.jenkinsci.test.acceptance.po;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+/**
+ * {@link PageObject} for the snippet generator to create bits of code for individual steps.
+ */
+public class SnippetGenerator extends PageObject {
+    private static final String URI = "pipeline-syntax/";
+
+    /**
+     * Creates a new page object for the snippet generator.
+     *
+     * @param context
+     *         job context
+     */
+    public SnippetGenerator(final WorkflowJob context) {
+        super(context,  context.url(URI));
+    }
+
+    @Override
+    protected WorkflowJob getContext() {
+        return (WorkflowJob) super.getContext();
+    }
+
+    /**
+     * Generates the sample pipeline script.
+     *
+     * @return the generated script
+     */
+    public String generateScript() {
+        WebElement button = find(By.id("generatePipelineScript-button"));
+        button.click();
+        elasticSleep(500);
```

Comment:
please avoid introducing new POs with elastic sleep where they can be avoided.

if the callers are never going to call generate script more than once you can wait `wait.until` the `textarea` is not empty.  (and add this fact to the javadoc)
if it is going to call it more than once with different options then it should be possible to get a before and wait until it is not the same.